interfaces/apparmor/spec_test.go

```diff
@@ -99,3 +99,36 @@ func (s *specSuite) TestSpecificationIface(c *C) {
 		"snap.snap2.app2": {"connected-slot", "permanent-slot"},
 	})
 }
+
+// AddSnippet adds a snippet for the given security tag.
+func (s *specSuite) TestAddSnippet(c *C) {
+	restore := apparmor.SetSpecScope(s.spec, []string{"snap.demo.demo", "snap.demo.service"}, "demo")
+	defer restore()
+
+	// Add two snippets in the context we are in.
+	s.spec.AddSnippet("snippet 1")
+	s.spec.AddSnippet("snippet 2")
+
+	// The snippets were recorded correctly.
+	c.Assert(s.spec.Snippets(), DeepEquals, map[string][]string{
+		"snap.demo.demo":    {"snippet 1", "snippet 2"},
+		"snap.demo.service": {"snippet 1", "snippet 2"},
+	})
+	c.Assert(s.spec.SnippetForTag("snap.demo.demo"), Equals, "snippet 1\nsnippet 2")
+	c.Assert(s.spec.SecurityTags(), DeepEquals, []string{"snap.demo.demo", "snap.demo.service"})
+}
+
+// AddSunSnippet adds a snippet for the snap-update-ns profile for a given snap.
+func (s *specSuite) TestAddSunSnippet(c *C) {
+	restore := apparmor.SetSpecScope(s.spec, []string{"snap.demo.demo", "snap.demo.service"}, "demo")
+	defer restore()
+
+	// Add a two snap-update-ns snippets in the context we are in.
+	s.spec.AddSunSnippet("snippet 1")
+	s.spec.AddSunSnippet("snippet 2")
+
+	// The snippets were recorded correctly.
+	c.Assert(s.spec.SunSnippets(), DeepEquals, map[string][]string{
+		"demo": {"snippet 1", "snippet 2"},
+	})
```

Comment:
Shouldn't you have a negative test that snap.demo.service doesn't have either "snippet 1" or "snippet 2"?

Reply:
Good idea, I'll add negative tests as well.